### PR TITLE
DDF-1442: Added BasicAuth iTests

### PR DESCRIPTION
**Pull request by emanns95**
Added integration tests to verify that BasicAuth is working for DDF clients. The tests set up sources and submit queries to ensure that given proper authentification the query can be completed, and with bad certification, it cannot.

@kcwire @jrnorth @kguerriero @jlcsmith  @millerw8 @beyelerb @wmcnalli 
Champ: @vinamartin

[Review on Reviewable](https://reviewable.io/reviews/codice/ddf/131)

### CODE REVIEW

**Comment by jrnorth** [comment]
Reviewed 4 of 4 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/131)

File: distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestSecurity.java [hunk 1]
```diff
@@ -136,6 +136,70 @@ public void testBasicRestAccess() throws Exception {
...
+                SERVICE_ROOT + "/catalog/query?q=*&src=" + unavailableOpenSourceId;
+
+        given().auth().basic("admin", "admin").when().get(unavailableOpenSearchQuery).then().log()
+                .all().assertThat().statusCode(equalTo(200)).assertThat()
+                .body(not(containsString("myTitle")));
+
+        String unavailableCswQuery = SERVICE_ROOT + "/catalog/query?q=*&src=" + unavailableCswSourceId;
+        */
+
+        configureRestForAnonymous();
```
Comment (emanns95) [inline distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestSecurity.java]: These two lines are having issues. We're having an issue deleting the metacard we ingest. We thought we had resolved the issue by making sure to reconfigure rest for anonymous before deleting the card, and it had worked once but now fails. Does configuring for anonymous require some time that we should wait for before attempting to delete the metacard? I'd like feedback from @stustison and @pklinef on this.

**Comment by kcwire** [comment]
Reviewed 4 of 4 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/131)

**Comment by emanns95** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

<sup>**[distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestSecurity.java, line 196 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/131#-JxC8pBaLbbp2pLMSpif)** ([raw file](https://github.com/codice/ddf/blob/79284a727923d57802537c5f7cfcf7bcfac69b83/distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestSecurity.java#L196)):</sup>
Whoops, I'll remove this later

---

<sup>**[distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestSecurity.java, line 199 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/131#-JxB7Jl3_MNSJLZO1auX-r1-199)** ([raw file](https://github.com/codice/ddf/blob/79284a727923d57802537c5f7cfcf7bcfac69b83/distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestSecurity.java#L199)):</sup>
So this is a known race condition that will need to be addressed in its own ticket. The tests will intermittently fail when trying to delete the metacard due to the configureRestForAnonymous not immediately working. The metacard must be deleted or it will affect other tests, so this is a blockage

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/131)

**Comment by emanns95** [comment]
@vinamartin go ahead now. @kcwire finished

File: distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestSecurity.java [hunk 2]
```diff
@@ -136,6 +136,75 @@ public void testBasicRestAccess() throws Exception {
...
+        /*
+        DDF-1442: Sleeps for a small time so that the web context policy can be updated and make authentication not be
+        required for the deletion of the metacard.
+         */
+        Thread.sleep(CONFIG_UPDATE_WAIT_INTERVAL);
```
Comment (emanns95) [inline distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestSecurity.java]: @pklinef

**Comment by kguerriero** [comment]
Reviewed 2 of 4 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks pending.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/131)

**Comment by vinamartin** [comment]
lgtm! :+1:

**Pull request merged**